cli/functionary/package.py

```diff
@@ -107,20 +102,17 @@ def publish(ctx, path, host):
             publish_url, headers=headers, files={"package_contents": upload_file}
         )
     except requests.ConnectionError:
-        click.echo(f"Unable to connect to {host}")
-        ctx.exit(2)
+        click.ClickException(f"Unable to connect to {host}")
     except requests.Timeout:
-        click.echo("Timeout occurred waiting for build")
-        ctx.exit(2)
+        click.ClickException("Timeout occurred waiting for build")
 
     # check status code/message on return then exit
     if upload_response.ok:
         click.echo("Build succeeded")
     else:
-        click.echo(
+        click.ClickException(
             f"Failed to build image: {upload_response.status_code}\n"
             f"\tResponse: {upload_response.text}"
         )
         if upload_response.status_code == 401:
-            click.echo("\n\nTry log in again.")
-        ctx.exit(1)
+            click.ClickException("\n\nTry log in again.")
```

Comment:
May as well fix the bad English here.  Also, I'm not sure that the flow here is going to actually work.  Presumably the `click.ClickException` above is support to be raised, which means you'd never reach the `if upload_response.status_code == 401` block.  Take another look at what this whole `else` block is trying to achieve and see if you can correctly rework it while using the `ClickException`.